planning/behavior_path_planner/src/util/lane_change/util.cpp

```diff
@@ -337,10 +337,13 @@ std::pair<bool, bool> getLaneChangePaths(
   LaneChangeTargetObjectIndices dynamic_object_indices;
 
   candidate_paths->reserve(lane_change_sampling_num);
-  for (double acceleration = 0.0; acceleration >= maximum_deceleration;
-       acceleration -= acceleration_resolution) {
+  for (double sampled_acc = 0.0; sampled_acc >= maximum_deceleration;
+       sampled_acc -= acceleration_resolution) {
     const auto prepare_speed =
-      std::max(current_velocity + acceleration * prepare_duration, minimum_lane_change_velocity);
+      std::max(current_velocity + sampled_acc * prepare_duration, minimum_lane_change_velocity);
+
+    // compute actual acceleration
+    const double acceleration = (prepare_speed - current_velocity) / prepare_duration;
```

Comment:
maximum deceleration is already defined as you suggested in the current code. You can check it from the following link.

https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/src/util/lane_change/util.cpp#L293-L300

Reply:
Therefore, there should be no meaning less loop in the current code.

Reply:
Oh I see, but in that case `acceleration` and `sample_acc` are always the same no?

Reply:
No, when the current velocity is smaller than minimum lane change velocity, samples acc and acceleration have different values

Reply:
You're right.
That said, I find the accel to velocity then to accel conversion suspiciously complex.

when `current_velocity <= minimum_lane_change_velocity` we have:
- `min_a >= 0` (the local `min_a` in the lambda)
- `maximum_deceleration` = `-epsilon`
- the `for` loop runs once with `sampled_acc = 0.0`. In that case `acceleration = min_a`

When `current_velocity > minimum_lane_change_velocity` we have:
- `min_a < 0`
- `-std::abs(parameter.maximum_deceleration) <= maximum_deceleration < -epsilon`
- the `for` loop loops with `acceleration` from 0.0 to `maximum_deceleration`. Since we have `min_a <= maximum_deceleration <= sample_acc`, then `current_velocity + min_a * prepare_duration = minimum_lane_change_velocity <= current_velocity + sampled_acc * prepare_duration`. Thus, `prepare_speed = current_velocity + sampled_acc * prepare_duration` and `acceleration = (prepare_speed - current_velocity) / prepare_duration = sampled_acc`

In summary, there is one case where the loop is executed only once with `acceleration = min_a >= 0`, and another case where `acceleration = sample_acc` in `[maximum_deceleration, 0.0]` (unless I made a mistake in my calculation).

Now, if we define `maximum_deceleration = std::max(min_a, -std::abs(parameter.maximum_deceleration))` (instead of the clamp), we get:
- when `maximum_deceleration > 0`, loop once with `accel = maximum_deceleration`
- otherwise, loop from `maximum_deceleration <= accel <= 0`.

So the for loop could be replaced by a simpler:
```C++
double accel = maximum_deceleration;
do {
    // do stuff
} while(accel -= acceleration_resolution, accel < 0.0);
```

I think it does the same than the version with `sample_acc`, except we don't need to the intermediate variables and many accel->vel->accel conversions. Although it is usually recommended not to use `do {} while();`, I find it is perfectly suited here (there is no "simple" way to write the same with a `for`)

Reply:
@VRichardJP As you said, we should not use `for` in this calculation which makes accel->vel->accel conversions. However, if I try to use `while`, there would occur many changes, and I intentionally use the intermediate variable right now. We are planning to clean the code next month, and I would change the `for` statement to `while` at the time.

Btw, is your code really correct? I feel this does not work properly. When `current_velocity < min_lane_change_speed`, does this really work as the current code?
```
double accel = maximum_deceleration; (isn't it 0.0 here?)
do {
    // do stuff
} while(accel -= acceleration_resolution, accel < 0.0);
```

Reply:
@VRichardJP Your code does not work for the current lane change module. Since lane change does not want to decelerate as much as possible, we should search for a lane change path from the acceleration = 0.0. On top of that, lane change module returns a lane change path as soon as it finds a safe and valid path in the for loop. Therefore, we have to search from the acceleration=0.0

I know your suggestion can make the code shorter, but it does not fit the current algorithm flow.